src/coreclr/vm/threads.cpp

```diff
@@ -7695,24 +7695,25 @@ InterpThreadContext* Thread::GetInterpThreadContext()
 {
     WRAPPER_NO_CONTRACT;
 
-    // HACK: FIXME: InterpreterStub can be called on a native thread with no managed Thread,
-    //  and when this happens it calls us with a null thisptr in order to populate everything
-    if (this == nullptr)
-    {
-        _ASSERTE(GetThreadNULLOk() == NULL);
-        Thread *actualThread = SetupThreadNoThrow();
-        if (actualThread == NULL)
-            COMPlusThrow(kOutOfMemoryException);
-        return actualThread->GetInterpThreadContext();
-    }
-
     if (m_pInterpThreadContext == nullptr)
     {
         m_pInterpThreadContext = new (nothrow) InterpThreadContext();
     }
 
     return m_pInterpThreadContext;
 }
+
+extern "C" InterpThreadContext* STDCALL GetInterpThreadContextWithPossiblyMissingThread(Thread *pThread)
+{
+    if (pThread == nullptr)
+    {
+        pThread = SetupThreadNoThrow();
+        if (pThread == NULL)
+            COMPlusThrow(kOutOfMemoryException);
```

Comment:
```suggestion
        pThread = SetupThread();
```
If we are going to throw, we can just call the throwing version